### ISSUE TITLE
Null ref: T2.DemonStatue.OnCreaturePressedActionToInteract

### ISSUE THREAD

**Issue by ChrisNZL** [ChrisNZL/Tallowmere2#740]
Auto bug report. v0.2.1. Online Co-op.

Feedback ID: 20201214-K6TK2

```
1:21:28, Frame 1631619, EXCEPTION »  NullReferenceException: Object reference not set to an instance of an object
>>>>> CRITICAL ERROR >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

T2.DemonStatue.OnCreaturePressedActionToInteract (T2.Creature creature)
T2.Interactable.OnCreaturePressedActionToInteract (T2.Creature creature)
T2.CreatureBody.Update ()

GameStates: InitialisingAsNetworkHost, RompingThroughDungeon
GameSetupMode: NetworkCoop
Players: 2
DungeonRoomTransitionerState: Null
Room: 3 / 3
RoomModifiers: None
YouHaveDiedState: Null
SystemPlayer InputDevice: XInput Controller
HumanPlayer 1 InputDevice: XInput Controller
```

**Comment by ChrisNZL** [comment]
This was fixed in 0.2.2a.

Added many null checks within `DemonStatue.OnCreaturePressedActionToInteract`.

**Comment by ChrisNZL** [comment]
Should be fixed in 0.3.1a.

Added and adjusted null checks within `DemonStatue.OnCreaturePressedActionToInteract`, especially checking against `CreatureStruct.pendingRewardSets` in case the list was null.
